cpp/zotero_enhancement_maps_generator.cc

```diff
@@ -0,0 +1,145 @@
+ *  \author Mario Trojan
+ *
+ *  \copyright 2020 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <iostream>
+#include <unordered_set>
+#include <cstdio>
+#include <cstdlib>
+#include "FileUtil.h"
+#include "IniFile.h"
+#include "MiscUtil.h"
+#include "StringUtil.h"
+#include "UBTools.h"
+#include "ZoteroHarvesterConfig.h"
+#include "ZoteroHarvesterUtil.h"
+#include "ZoteroHarvesterZederInterop.h"
+#include "util.h"
+
+
+// avoid conflicts with LICENSE defined in mysql_version.h
+#ifdef LICENSE
+    #undef LICENSE
+#endif
+
+
+namespace {
+
+
+using namespace ZoteroHarvester;
+
+
+[[noreturn]] void Usage() {
+    ::Usage("[options] [zotero_enhancement_maps_directory]\n\n"
+            "\n"
+            "\tOptions:\n"
+            "\t[--min-log-level=log_level]          Possible log levels are ERROR, WARNING (default), INFO and DEBUG\n"
+            "\t[zotero_enhancement_maps_directory]  or default " + UBTools::GetTuelibPath() + "zotero-enhancement-maps" + "\n");
+}
+
+
+void DownloadZederInstanceEntries(const Zeder::Flavour flavour,
+                                  std::unordered_map<Zeder::Flavour, Zeder::EntryCollection> * const downloaded_entries)
+{
+    LOG_INFO("Downloading Zeder entries for " + Zeder::FLAVOUR_TO_STRING_MAP.at(flavour));
+    Zeder::EntryCollection downloaded_flavour_entries;
+
+    const auto endpoint_url(Zeder::GetFullDumpEndpointPath(flavour));
+    const std::unordered_set<unsigned> &entries_to_download {}; // intentionally empty
+    const std::unordered_set<std::string> columns_to_download {};  // intentionally empty
+    const std::unordered_map<std::string, std::string> filter_regexps {}; // intentionally empty
+    std::unique_ptr<Zeder::FullDumpDownloader::Params> downloader_params(new Zeder::FullDumpDownloader::Params(endpoint_url,
+                                                                         entries_to_download, columns_to_download, filter_regexps));
+
+    auto downloader(Zeder::FullDumpDownloader::Factory(Zeder::FullDumpDownloader::Type::FULL_DUMP, std::move(downloader_params)));
+    if (not downloader->download(&downloaded_flavour_entries))
+        LOG_ERROR("couldn't download full dump for " + Zeder::FLAVOUR_TO_STRING_MAP.at(flavour));
```

Comment:
Please include the error message of the downloader in parentheses.

Reply:
Zeder::FullDumpDownloader does not provide an error message (this is NOT from Downloader.h)